models/ProfileModel.js

```diff
@@ -122,27 +122,58 @@ const getUserLikes = `
   )
   ORDER BY p.timestamp DESC`;
 
-const getProfileContents = `
-  SELECT 
-    (SELECT COUNT(*) FROM
-      post WHERE "userId" = a."userId"
-    ) AS "postCount",
-    a."bio",
-    a."joinedDate",
-    a."displayName",
-    a."username",
-    (SELECT COUNT(*) FROM 
-      follow WHERE "followedUserId" = a."userId"
-    ) AS "followerCount",
-    (SELECT COUNT(*) FROM 
-      follow WHERE "followerUserId" = a."userId"
-    ) AS "followingCount"
-  FROM app_user AS a
-  WHERE a."username" = $1`;
+const getProfileContents = `SELECT 
+(SELECT COUNT(*) FROM post WHERE "userId" = a."userId") AS "postCount",
+a."bio",
+a."joinedDate",
+a."displayName",
+a."username",
+(SELECT COUNT(*) FROM follow WHERE "followedUserId" = a."userId" AND "followStatus" = TRUE ) AS "followerCount",
+(SELECT COUNT(*) FROM follow WHERE "followerUserId" = a."userId" AND "followStatus" = TRUE ) AS "followingCount",
+f."followStatus" AS "followStatus"
+FROM app_user AS a
+LEFT JOIN follow as f ON f."followerUserId" = $1 AND f."followedUserId" = a."userId"
+WHERE a."username" = $2`;
+
+const getFollowStatus = `SELECT 
+CASE 
+    WHEN EXISTS (
+        SELECT 1
+        FROM follow
+        WHERE "followerUserId" = $1
+          AND "followedUserId" = (SELECT "userId" FROM app_user WHERE "username" = $2)
+          AND "followStatus" = TRUE
+    ) THEN TRUE
+    ELSE FALSE
+END AS "followStatus";`;
+
+const followUser = `INSERT INTO follow ("followerUserId", "followedUserId", "followedDate", "followStatus")
+SELECT $1, "userId", CURRENT_DATE, TRUE
+FROM app_user
+WHERE "username" = $2
+ON CONFLICT ("followerUserId", "followedUserId")
+DO UPDATE SET "followStatus" = TRUE, "followedDate" = CURRENT_DATE
+RETURNING *;
+`;
+
+const unfollowUser = `WITH otherUser AS (
+SELECT "userId"
+FROM app_user
+WHERE "username" = $2
+LIMIT 1
+)
+UPDATE follow
+SET "followStatus" = FALSE
+WHERE "followerUserId" = $1
+AND "followedUserId" = (SELECT "userId" FROM otherUser)
```

Comment:
**Mentioned in one of the other requested changes:** "I don't want to do hard deletes because that seems like bad practice and it's almost like we are losing something that previously occurred in our application. Soft deletes are not ideal either but we could shift to "event" based tables, almost transaction-like. If someone unfollows, they append to a table that tracks events for unfollows, etc.".

We can discuss this further though

Reply:
After further discussion offline we'll be switching to hard deletes because losing this data shouldn't be a huge deal for our project. We likely won't use past following data, so hard deleting is fine